edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/metrics/AvaliabilityMetrics.cs

```diff
@@ -114,16 +138,10 @@ TimeSpan CalculateEdgeAgentDowntime()
             return TimeSpan.Zero;
         }
 
-        void NoteCurrentTime(object sender, EventArgs e)
+        Task NoteCurrentTime()
         {
-            try
-            {
-                File.WriteAllText("shutdown_time", this.time.UtcNow.Ticks.ToString());
-            }
-            catch (Exception ex)
-            {
-                this.log.LogError($"Could not save shutdown time:\n{ex}");
-            }
+            File.WriteAllText("avaliability_checkpoint", this.time.UtcNow.Ticks.ToString());
```

Comment:
This file is written to the container; it will be lost if a container is re-created.  Is this expected?

Reply:
yes